### PR TITLE
[REF254] Address edge case of single cross section induced script generator failure 

### PR DESCRIPTION
**Pull request by KedoKudo**
Original Gitlab Task: [REF254](https://code.ornl.gov/sns-hfir-scse/reflectometry/reflectometry/-/issues/254)

This PR introduces a quick patch to deal with an edge case where a run with single cross section leads to the failure of script generation.
To test, please follow the instruction video in the parent defect [REF244](https://code.ornl.gov/sns-hfir-scse/reflectometry/reflectometry/-/issues/244).
The resulting output should be something similar to the screenshots below

- QuickNXS main GUI: 
image
- Terminal (no error messages, only warnings):
image
- Generated files listed in terminal
image

### CODE REVIEW

File: reflectivity_ui/interfaces/data_handling/data_manipulation.py [hunk 1]
```diff
@@ -40,21 +40,34 @@ def generate_short_script(reduction_list):
             continue
 
         ws_name = "r%s" % reduction_list[i].cross_sections[xs].number
-        if len(api.mtd[ws_name]) == 0:
-            logging.info("  No entry in %s workspace group", ws_name)
+
+        # NOTE: It is possible that only one cross section is present in the run, therefore
+        #       api.mtd[ws_name] could be a Workspace2D instead of a workspcae group.
```
Comment (peterfpeterson) [inline reflectivity_ui/interfaces/data_handling/data_manipulation.py:45]:
```suggestion
        #       api.mtd[ws_name] could be a Workspace2D instead of a workspace group.
```

**Review by peterfpeterson — CHANGES_REQUESTED** [review]

File: reflectivity_ui/interfaces/data_handling/data_manipulation.py [hunk 2]
```diff
@@ -40,21 +40,34 @@ def generate_short_script(reduction_list):
...
+
+        # NOTE: It is possible that only one cross section is present in the run, therefore
+        #       api.mtd[ws_name] could be a Workspace2D instead of a workspace group.
+        #       Given that a modernization process is scheduled, we are going to do a ducktape
+        #       fixing here.
+        contain_single_crosssection = False
+        try:
+            if len(api.mtd[ws_name]) == 0:
+                logging.info("  No entry in %s workspace group", ws_name)
+        except:
```
Comment (peterfpeterson) [inline reflectivity_ui/interfaces/data_handling/data_manipulation.py:52]: Bare exceptions are a bit dangerous. Change this to `except TypeError` and add a comment about assuming the object is actually a Workspace2D.

Comment (KedoKudo) [reply reflectivity_ui/interfaces/data_handling/data_manipulation.py]: The exception handling is adjusted as suggested.

**Review by peterfpeterson — APPROVED** [review]
